be/src/exec/es_http_scan_node.cpp

```diff
@@ -0,0 +1,445 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/es_http_scan_node.h"
+
+#include <chrono>
+#include <sstream>
+
+#include "common/object_pool.h"
+#include "exec/es/es_predicate.h"
+#include "exec/es/es_query_builder.h"
+#include "exec/es/es_scan_reader.h"
+#include "exec/es/es_scroll_query.h"
+#include "exprs/expr.h"
+#include "runtime/runtime_state.h"
+#include "runtime/row_batch.h"
+#include "runtime/dpp_sink_internal.h"
+#include "service/backend_options.h"
+#include "util/runtime_profile.h"
+
+namespace doris {
+
+EsHttpScanNode::EsHttpScanNode(
+        ObjectPool* pool, const TPlanNode& tnode, const DescriptorTbl& descs) : 
+            ScanNode(pool, tnode, descs), 
+            _tuple_id(tnode.es_scan_node.tuple_id),
+            _runtime_state(nullptr),
+            _tuple_desc(nullptr),
+            _num_running_scanners(0),
+            _scan_finished(false),
+            _eos(false),
+            _max_buffered_batches(1024),
+            _wait_scanner_timer(nullptr) {
+}
+
+EsHttpScanNode::~EsHttpScanNode() {
+}
+
+Status EsHttpScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
+    RETURN_IF_ERROR(ScanNode::init(tnode));
+
+    // use TEsScanNode
+    _properties = tnode.es_scan_node.properties;
+    return Status::OK;
+}
+
+Status EsHttpScanNode::prepare(RuntimeState* state) {
+    VLOG_QUERY << "EsHttpScanNode prepare";
+    RETURN_IF_ERROR(ScanNode::prepare(state));
+
+    _runtime_state = state;
+    _tuple_desc = state->desc_tbl().get_tuple_descriptor(_tuple_id);
+    if (_tuple_desc == nullptr) {
+        std::stringstream ss;
+        ss << "Failed to get tuple descriptor, _tuple_id=" << _tuple_id;
+        return Status(ss.str());
+    }
+
+    // set up column name vector for ESScrollQueryBuilder
+    for (auto slot_desc : _tuple_desc->slots()) {
+        if (!slot_desc->is_materialized()) {
+            continue;
```

Comment:
This should add a warning log, because the tuple received by BE does't contains unmaterialized slot in normal condition, except it's a bug.

Reply:
sometimes slot is not materialized, for example in query `select a from (select * from xxx) yyy`

Reply:
No, it will materialize a in xxx.

Reply:
but other column is not materialized